packages/lit-html/src/directives/ref.ts

```diff
@@ -38,7 +38,7 @@ const lastElementForContextAndCallback: WeakMap<
   WeakMap<Function, Element | undefined>
 > = new WeakMap();
 
-export type RefOrCallback = Ref | ((el: Element | undefined) => void);
+export type RefOrCallback<T = Element> = Ref<T> | ((el: T | undefined) => void);
```

Comment:
Should T extend Element?

Reply:
Ref objects and callbacks could _technically_ be used with any type. `Ref` already doesn't have T extending Element, so this also seems inline with that.